pkg/actions/nodegroup/ami.go

```diff
@@ -0,0 +1,31 @@
+package nodegroup
+
+import (
+	"strings"
+
+	"github.com/blang/semver"
+)
+
+type amiReleaseVersion struct {
+	Version semver.Version
+	Date    string
+}
+
+// LTE checks if a is less than or equal to b.
+func (a amiReleaseVersion) LTE(b amiReleaseVersion) bool {
+	return a.Compare(b) <= 0
+}
+
+// GTE checks if a is greater than or equal to b.
+func (a amiReleaseVersion) GTE(b amiReleaseVersion) bool {
+	return a.Compare(b) >= 0
+}
+
+// Compare returns 0 if a==b, -1 if a < b, and +1 if a > b.
```

Comment:
This description should include that it first checks the version and if those equal, THEN it will do the `a == b` ... etc things.